### DISCUSSION TITLE
Unable to send mail. Please try again later

### DISCUSSION THREAD

**Discussion by magneto0007** [wardenenv/warden#598] [Help & Support]
Hi

I've installed Magento2 successfully, when I try to send an email using contact form via Magento frontend, it gives  **Unable to send mail. Please try again later.** error.

Mailhog as well has no emails listed

Note: Using MacBook M2

Any help please ?

**Comment by sprankhub** [comment]
I suggest checking if you have any SMTP module enabled and configured.

**Comment by magneto0007** [comment]
@sprankhub thanks for your reply. This is a deafult warden build so Im expecting it to work.

I can see the below error

> php -r 'mail("test@example.com", "test","test");'
> runtime: failed to create new OS thread (have 2 already; errno=22)
> fatal error: newosproc
> 
> runtime stack:
> runtime.throw(0x6b2000, 0x9)
> 	/usr/local/Cellar/go/1.6/libexec/src/runtime/panic.go:530 +0x90
> runtime.newosproc(0xc82002a000, 0xc820039fc0)
> 	/usr/local/Cellar/go/1.6/libexec/src/runtime/os1_linux.go:149 +0x18c
> runtime.newm(0x715c28, 0x0)
> 	/usr/local/Cellar/go/1.6/libexec/src/runtime/proc.go:1513 +0x135
> runtime.main.func1()
> 	/usr/local/Cellar/go/1.6/libexec/src/runtime/proc.go:125 +0x2c
> runtime.systemstack(0x7f3300)
> 	/usr/local/Cellar/go/1.6/libexec/src/runtime/asm_amd64.s:291 +0x79
> runtime.mstart()
> 	/usr/local/Cellar/go/1.6/libexec/src/runtime/proc.go:1048
> 
> goroutine 1 [running]:
> runtime.systemstack_switch()
> 	/usr/local/Cellar/go/1.6/libexec/src/runtime/asm_amd64.s:245 fp=0xc820024770 sp=0xc820024768
> runtime.main()
> 	/usr/local/Cellar/go/1.6/libexec/src/runtime/proc.go:126 +0x62 fp=0xc8200247c0 sp=0xc820024770
> runtime.goexit()
> 	/usr/local/Cellar/go/1.6/libexec/src/runtime/asm_amd64.s:1998 +0x1 fp=0xc8200247c8 sp=0xc8200247c0

Might be something to do with using Macbook Silicon ?

**Reply by navarr** [reply]
This looks like you're running it on the host machine instead of inside the `warden shell`?

**Reply by magneto0007** [reply]
@navarr running in **warden shell**

**Comment by magneto0007** [comment]
Ok, I switched from Warden to Den for Macbook Silicon and now it works.

Here a link to Den https://swiftotter.github.io/den/installing.html

**Comment by iranimij** [comment]
I have the same problem @navarr Can't we fix this issue in Warden? I think it happened after updating to the latest version of the warden. I also have macBook.

**Reply by navarr** [reply]
Can you provide the output of `warden env config`, with anything you need to hide (like project names or private images) redacted?

**Reply by iranimij** [reply]
[config.txt](https://github.com/user-attachments/files/18331446/config.txt)
I attached the config, @navarr Please check.
